0.data_analysis_and_processing/utils/PairwiseCompare.py

```diff
@@ -0,0 +1,196 @@
+import warnings
+from collections.abc import Iterable
+from itertools import combinations, product
+from typing import Optional
+
+import pandas as pd
+from Comparator import Comparator
+
+
+class PairwiseCompare:
+    """
+    Compute inter or intra group comparisons between objects.
+    """
+
+    def __init__(
+        self,
+        _df: pd.DataFrame,
+        _comparator: Comparator,
+        _antehoc_group_cols: list[str],
+        _posthoc_group_cols: list[str],
+        _feat_cols: list[str],
+        _drop_cols: Optional[list[str]] = None
+    ):
+        """
+        Perform Input validation and initializations.
+
+        Parameters
+        ----------
+        _df:
+            Contains the features and group columns to use for comparing paired data.
+
+        _comparator:
+            Defines how to perform and save comparisons.
+
+        _antehoc_group_cols:
+            Columns to groupby before organizing by _posthoc_group_cols.
+
+        _feat_cols:
+            Feature columns for creating comparisons.
+
+        _posthoc_group_cols:
+            Columns to groupby after organizing by _antehoc_group_cols.
```

Comment:
I am a bit confused by what groupby here does? Is this after computing the comparisons?